docs/root/updating_envoy_dependency.md

```diff
@@ -0,0 +1,463 @@
+      ```
+      ENVOY_COMMIT = "9753819331d1547c4b8294546a6461a3777958f5"
+      ENVOY_SHA = ""
+      ```
+1. Run `ci/do_ci.sh build`, notice the sha256 value at the top of the output,
+   example:
+      ```
+      INFO: SHA256 (https://github.com/envoyproxy/envoy/archive/9753819331d1547c4b8294546a6461a3777958f5.tar.gz) = f4d26c7e78c0a478d959ea8bc877f260d4658a8b44e294e3a400f20ad44d41a3
+      ```
+1. Update `ENVOY_SHA` in [bazel/repositories.bzl](bazel/repositories.bzl) to
+      this value.
+
+#### Example commands
+
+Overwrite `ENVOY_COMMIT` and `ENVOY_SHA` in `bazel/repositories.bzl`:
+
+```bash
+sed -i -e "s/ENVOY_COMMIT =.*/ENVOY_COMMIT = \"${envoy_commit}\"/" bazel/repositories.bzl
+
+sed -i -e "s/ENVOY_SHA =.*/ENVOY_SHA = \"\"/" bazel/repositories.bzl
+
+git diff
+```
+
+At this point:
+
+- `ENVOY_COMMIT` should be a new value
+- `ENVOY_SHA` should be blank
+
+Obtain the new Envoy SHA by running the first part of a build:
+
+```bash
+bazel clean --expunge
+
+envoy_sha=$(ci/do_ci.sh build 2>&1 | head -30 | egrep -m 1 -o '[0-9a-f]{64}')
+
+echo "envoy_sha=$envoy_sha"
+```
+
+If we successfully obtained the new SHA, set `ENVOY_SHA`:
+
+```bash
+sed -i -e "s/ENVOY_SHA =.*/ENVOY_SHA = \"$envoy_sha\"/" bazel/repositories.bzl
+
+git diff
+```
+
+At this point:
+
+- `ENVOY_COMMIT` should be a new value
+- `ENVOY_SHA` should be a new value
+
+### Step 5
+
+Set up a Bash function `merge_from_envoy` that will be used repeatedly in the example commands in steps 6-10.
+
+Paste the following into the shell:
+
+```bash
+merge_from_envoy() {
+  # $1 = relative path to Nighthawk file derived from a version in the Envoy repo
+  relpath=$1
+
+  pushd $envoy_dir
+  git log $relpath | head
+  popd
+
+  diff $envoy_dir/$relpath $relpath || true
+
+  echo "Determine if $relpath needs to be manually merged:"
+  echo "  Any differences not marked with '# unique'?"
+  echo "  If so, edit $relpath in another terminal before continuing."
+}
+```
+
+This will only work in a shell where `$envoy_dir` was previously set and we
+cloned Envoy locally (see Step 3).
+
+When running this function, in the terminal you will see:
+
+- commit id of Envoy's version of the file (e.g. c9d883afbd9bf5046f6bb6dbfab724bbcc104123)
+- a diff of Envoy (left) and Nighthawk (right)
+
+Our goal is to transfer changes from Envoy to Nighthawk, preserving certain Nighthawk-specific changes marked with `#unique`.
+
+To take a closer look at an Envoy commit, visit `https://github.com/envoyproxy/envoy/commit/INSERT_COMMIT_ID_HERE`. This is helpful when it's hard to tell from the diff how the file should look.
+
+Updates in the Envoy file should be selectively pasted into the Nighthawk file using a text editor, possibly in a second terminal.
+
+Once you have done some partial work, save the file and repeat the `merge_from_envoy` command in the original terminal to check the diff again. This can be done iteratively.
+
+### Step 6
+
+Sync (copy) [.bazelrc](.bazelrc) from
```

Comment:
The `.bazelrc` link doesn't work. We will probably have the same problem on multiple links. Please verify the rest of the links.

Reply:
Fixed all links and confirmed that all links in the latest revision can be opened.